api/src/models/user_program.rs

```diff
@@ -9,7 +9,16 @@ use diesel::{
 use uuid::Uuid;
 
 /// Expression to filter user_programs by owner's username and program spec ID
-type WithUserAndProgramSpec<'a> = dsl::And<
+type WithUserAndProgramSpec = dsl::And<
+    dsl::Eq<user_programs::columns::user_id, Bound<sql_types::Uuid, Uuid>>,
+    dsl::Eq<
+        user_programs::columns::program_spec_id,
+        Bound<sql_types::Uuid, Uuid>,
+    >,
+>;
```

Comment:
Can you make this be called with user id, its a little confusing since the other one says username. Also comment seems out of date